src/main/java/jenkins/plugins/jobcacher/pipeline/CacheStepExecution.java

```diff
@@ -58,7 +58,7 @@ private void execute() throws Exception {
                 .start();
     }
 
-    private static class ExecutionCallback extends BodyExecutionCallback {
+    private class ExecutionCallback extends BodyExecutionCallback {
```

Comment:
Note that this is an incompatible change (would break running builds across upgrade). Better to leave the original class, deprecated, and introduce a new callback class.

Reply:
Thanks for pointing out, i will change it accordingly.

Reply:
I introduced a new class as proposed. I did not deprecate the old one though as I was not sure if adding an annotation will lead to a breaking change too.